module/epochs/qc_voter.go

```diff
@@ -170,13 +172,29 @@ func (voter *RootQCVoter) Vote(ctx context.Context, epoch protocol.Epoch) error
 		// update our last successful client index for future calls
 		voter.updateLastSuccessfulClient(clientIndex)
 		return nil
+	}
+	err = retry.Do(ctx, backoff, func(ctx context.Context) error {
```

Comment:
have you considered restructuring the code such that we can abort the voting by cancelling the context?